internal/pkg/cli/recorder/recorder.go

```diff
@@ -95,6 +95,22 @@ func (r *Recorder) Run() error {
 		ch := make(chan os.Signal, 1)
 		r.Notify(ch, os.Interrupt)
 		log.Print(WaitForSigIntMessage)
+
+		// searching the mntns of the process started outside of spoc
+		cmd := r.options.commandOptions.Command()
+		if err := util.Retry(func() (err error) {
```

Comment:
For seccomp, it currently works as follows if `--no-proc` is passed:

1. We record syscalls for all mount namespaces by binaries [with the target program name](https://github.com/kubernetes-sigs/security-profiles-operator/blob/32fe817cea4b121d248df110f99eba267af7f9dc/internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c#L126-L131).
2. We merge them below in processSeccomp.

Is there a reason why we don't want to do the same for AppArmor, i.e. merge all recorded profiles?

Reply:
My understanding of this flag is that the process is not started by the spoc CLI but externally and spoc will start recording the profile for it as soon as the PID comes live.

I think in the previous implementation,  it was recording everything for all process running into the system until it got interrupted. 

In this implementation, it will look up first the PID by command name, and then the namespace where that PID is running. This is required to retrieve the recorded information from the maps by namespace.

Now we create a apparmor profile for an entire namespace where the process is running. This is also the case for seccomp. This is not ideal, if other process run in parallel in that namespace when the CLI is recording. The profiles will contain more stuff than expected. I think for CLI this is acceptable, since people will most likely use this feature to record profiles for a single process running into a wider system, and that process will have its own namesapce unless they will start it in a dedicated namesapce. 

I would argue that this feature is probably not the best option, if you want to  record a profile for an entire container  (e.g. form outside). In that case, I would use the in-cluster option.

Reply:
> I think in the previous implementation, it was recording everything for all process running into the system until it got interrupted.

Not quite. The ebpf code itself [checks](https://github.com/kubernetes-sigs/security-profiles-operator/blob/32fe817cea4b121d248df110f99eba267af7f9dc/internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c#L126-L131) the program name and only records events for matching processes. So yes, you would get all events for "curl", no matter from which namespace, but you would not get events from the rest of the system. This is why the seccomp implementation [takes all namespaces into account](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/internal/pkg/cli/recorder/recorder.go#L183) (mntns is 0) with `--no-proc`. I think we could do the same for AppArmor and avoid the PID race.